packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js

```diff
@@ -45,29 +45,50 @@ function UploadFonts() {
 	 * @param {Array} files The files to be filtered
 	 * @return {void}
 	 */
-	const handleFilesUpload = ( files ) => {
+	const handleFilesUpload = async ( files ) => {
 		setNotice( null );
 		setIsUploading( true );
 		const uniqueFilenames = new Set();
 		const selectedFiles = [ ...files ];
-		const allowedFiles = selectedFiles.filter( ( file ) => {
+		let hasInvalidFiles = false;
+
+		// Use map to create a promise for each file check, then filter with Promise.all
+		const checkFilesPromises = selectedFiles.map( async ( file ) => {
+			const isFont = await isFontFile( file );
+			if ( ! isFont ) {
+				hasInvalidFiles = true;
+				return null; // Return null for invalid files
+			}
+			// Check for duplicates
 			if ( uniqueFilenames.has( file.name ) ) {
-				return false; // Discard duplicates
+				return null; // Return null for duplicates
 			}
-			// Eliminates files that are not allowed
+			// Check if the file extension is allowed
 			const fileExtension = file.name.split( '.' ).pop().toLowerCase();
 			if ( ALLOWED_FILE_EXTENSIONS.includes( fileExtension ) ) {
 				uniqueFilenames.add( file.name );
-				return true; // Keep file if the extension is allowed
+				return file; // Return the file if it passes all checks
 			}
-			return false; // Discard file extension not allowed
+			return null; // Return null for disallowed file extensions
 		} );
+
+		// Filter out the nulls after all promises have resolved
+		const allowedFiles = ( await Promise.all( checkFilesPromises ) ).filter(
+			( file ) => file !== null
+		);
+
 		if ( allowedFiles.length > 0 ) {
 			loadFiles( allowedFiles );
 		} else {
+			const message = hasInvalidFiles
+				? __(
+						'One or more of the selected files are not valid fonts. Please, try again.'
```

Comment:
Nit :) All other occurrences of "Please try again" in the plugin and in core does not use a comma between please and try.